core/src/main/java/org/apache/iceberg/encryption/PlaintextEncryptionManager.java

```diff
@@ -18,18 +18,22 @@
  */
 package org.apache.iceberg.encryption;
 
-import java.nio.ByteBuffer;
 import org.apache.iceberg.io.InputFile;
 import org.apache.iceberg.io.OutputFile;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 public class PlaintextEncryptionManager implements EncryptionManager {
+  private static final EncryptionManager INSTANCE = new PlaintextEncryptionManager();
   private static final Logger LOG = LoggerFactory.getLogger(PlaintextEncryptionManager.class);
 
+  public static EncryptionManager instance() {
+    return INSTANCE;
+  }
+
   @Override
   public InputFile decrypt(EncryptedInputFile encrypted) {
-    if (encrypted.keyMetadata().buffer() != null) {
+    if (encrypted.keyMetadata() != null && encrypted.keyMetadata().buffer() != null) {
       LOG.warn(
           "File encryption key metadata is present, but currently using PlaintextEncryptionManager.");
```

Comment:
Rather than using a class name, can we change this to "but no encryption has been configured"